app/src/main/java/com/example/habitsmasher/DaysTracker.java

```diff
@@ -0,0 +1,261 @@
+     *
+     * @param days: The string of dates to use, formatted according to above guide.
+     */
+    public DaysTracker(String days){
+
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+        Collections.fill(_days, Boolean.FALSE);
+
+        //take the string and split it up based on what days are present
+        //this will go under the assumption that the string will always be formatted correctly.
+        days = days.toUpperCase();
+        String[] splitDays = days.split(" ");
+
+        //for each item split from the string
+        for(int i = 0; i < splitDays.length; i++){
+            //get the value, and if any matches with the abbreviations set the corresponding day to true
+            String value = splitDays[i];
+            switch (value){
+                case "MO":
+                    _days.set(0, true);
+                    break;
+                case "TU":
+                    _days.set(1, true);
+                    break;
+                case "WE":
+                    _days.set(2, true);
+                    break;
+                case "TH":
+                    _days.set(3, true);
+                    break;
+                case "FR":
+                    _days.set(4, true);
+                    break;
+                case "SA":
+                    _days.set(5, true);
+                    break;
+                case "SU":
+                    _days.set(6, true);
+                    break;
+            }
+        }
+
+    }
+
+    /* Was experimenting with hashmaps and iterators, if anyone thinks this may be useful I will complete it,
+     * If not I'll get rid of it.
+
+    public DaysTracker(HashMap<String, Boolean> map){
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+        Collections.fill(_days, Boolean.FALSE);
+        Set<String> allKeys = map.keySet();
+
+        //iterator over key set
+        Iterator<String> keyIterator = allKeys.iterator();
+        while(keyIterator.hasNext()){
+            switch(keyIterator.next()){
+
+            }
+        }
+    }
+     */
+
+    /**
+     * Sets all of the days to false. Easier way to make a new DaysTracker without making a new DaysTracker.
+     */
+    public void setFalse(){
+        Collections.fill(_days, Boolean.FALSE);
+    }
+
+    /**
+     * Sets all of the days to true. Ease of use method for quickly setting all days to true.
+     */
+    public void setTrue(){
+        Collections.fill(_days, Boolean.TRUE);
+    }
+
+    /**
+     * Takes all of the days set to true and adds their abbreviations to a {@link String}, then returns the {@link String}.
+     * The string takes the format "day1 day2 ... day7", where each day is the first 2 letters of that day, separated by a space. The string can be in any order.
+     * EX) "MO FR WE TH"
+     *
+     * @return days: String containing all of the days set to true.
+     */
+    public String getDays(){
```

Comment:
This method is how days are recorded outside of the object. The string is stored in the days field in the DB, and can be interpreted in the constructor when passed.